File: CaesarCipher/Encryption/CaesarCipher.py
import string


def return_enc_key(i: str, lst: list[str], key: int) -> int:
    return (lst.index(i) + key) % 26


def main():
    alphabets = list(string.ascii_uppercase)
    newword = []
    key = int(input("Enter the Key: "))
    if 1 <= key <= 25:
        word = input("Enter the Word: ")
        for i in word.upper():
            if i  in alphabets:
                enc_key = return_enc_key(i, alphabets, key)
                newword.append(alphabets[enc_key])
        print("Encrypted Word: " + "".join(newword))
    else:
        print("Enter a valid key")


main()

File: CaesarCipher/Encryption/CaesarCipher.scala
def ReturnEncKey(num: Int, key: Int) : Int = {
	return (num+key)%26;
}

def ReadPlainText(key : Int) : Unit = {
	val stringBuilder = StringBuilder()
	val alphabets = ('A' to 'Z').zipWithIndex
	print("Enter the plaintext: ");
	var word = scala.io.StdIn.readLine();
	for (char <- word.toUpperCase.toList) {
		for ((alph, index) <- alphabets) {
			if (alph==char) {
				var enc_key = ReturnEncKey(index, key)
				var enc_char = alphabets(enc_key)._1
				stringBuilder.append(enc_char)
			}
		}
	}
	val resultString: String = stringBuilder.toString
	print(s"Encrypted Text: $resultString")
}

@main def main(): Unit =
	print("Enter Key: ")
	var key = scala.io.StdIn.readInt();
	//print(key);
	if (1<=key && key<=25) {
		ReadPlainText(key)		
	}
	else {
		print("Invalid Key")
	}
